app/models/course/assessment.rb

```diff
@@ -131,8 +131,13 @@ def validate_presence_of_questions
   end
 
   def validate_only_autograded_questions
-    return if questions.all?(&:auto_gradable?)
-    errors.add(:base, :autograded)
+    return unless published?
```

Comment:
Small issue but we can either put this at the validation - `validate :validate_ony_autograded_question, if: published?` or make the method name a little more meaningful, eg. `validate_only_autograded_question_published_assessment`.